python/ray/serve/tests/test_api.py

```diff
@@ -514,6 +515,37 @@ def f():
     serve.create_endpoint("endpoint", backend="backend")
 
 
+def test_create_infeasible_error(serve_instance):
+    serve.init()
+
+    def f():
+        pass
+
+    # Non existent resource should be infeasible.
+    with pytest.raises(RayServeException, match="Cannot scale backend"):
+        serve.create_backend(
+            "f:1",
+            f,
+            ray_actor_options={"resources": {
+                "MagicMLResource": 100
+            }})
+
+    # Even each replica might be feasible, the total might not be.
```

Comment:
We should also check one where the total resources are sufficient but the resource shape isn't (e.g., two nodes with 10 GPUs each, schedule an actor that requires 20 GPUs).

Reply:
checked in unit test